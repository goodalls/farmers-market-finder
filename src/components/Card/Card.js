import React, { Component } from 'react';
import MapContainer from '../MapContainer/MapContainer';
import * as actions from '../../actions/actions';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Card.css';

export class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      market: {},
      favorites: []
    };
  }
  componentDidMount() {
    this.setState({favorites: this.props.user});
  }

  favorite = (event, market) => {
    if (market.favorite === false) {
      market.favorite = true;
      this.props.updateFavorites(market);
      this.setState({market});
    } else {
      market.favorite = false;
      this.props.removeFavorite(market);
      this.setState({market: {}});
    }
  };

  render() {
    const { id, markets } = this.props;
    const marketInfo = markets.find(market => market.id === id);
    const schedule = marketInfo.Schedule.slice(0, -16);
    const products = marketInfo.Products.split(';').map((product, index) => {
      return <li key={index}>{product}</li>;
    });

    return (
      <div className="text-card">
        <span
          className={marketInfo.favorite ? 'favorite active' : 'favorite'}
          onClick={event => this.favorite(event, marketInfo)}
        >
          &#9829;
        </span>
        <h2 className="name">{marketInfo.marketname}</h2>
        <div className="info">
          <p>Address: {marketInfo.Address}</p>
          <p>Schedule: {schedule}</p>
          <ol>
            Products:
            {products}
          </ol>
        </div>
        <MapContainer address={marketInfo.Address} />
      </div>
    );
  }
}
Card.propTypes = {
  id: PropTypes.string,
  markets: PropTypes.array,
  fav: PropTypes.func,
  user: PropTypes.array,
  updateFavorites: PropTypes.func,
  removeFavorite: PropTypes.func
};

export const mapStateToProps = state => ({
  markets: state.markets,
  user: state.user.favorites
});

export const mapDispatchToProps = dispatch => ({
  updateFavorites: favorite => dispatch(actions.updateFavorites(favorite)),
  removeFavorite: favorite => dispatch(actions.removeFavorite(favorite))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Card));
