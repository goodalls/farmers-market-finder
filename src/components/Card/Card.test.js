import React from 'react';
import { shallow } from 'enzyme';
import { Card, mapStateToProps, mapDispatchToProps } from './Card';

describe('CARD', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Card markets={[{ Products: 'hello; big; string', Schedule: 'another large string'}]} user={[]} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('MSTP', () => {
    it('should map props from store', () => {
      const mockStore = { markets: [{}, {}], user: {favorites: []} };
      const mapped = mapStateToProps(mockStore);
      const expected = [{}, {}];
      expect(mapped.markets).toEqual(expected);
    });

    it('should map props from store', () => {
      const mockStore = { markets: [{}, {}], user: {favorites: []} };
      const mapped = mapStateToProps(mockStore);
      const expected = [];
      expect(mapped.user).toEqual(expected);
    });
  });
});
