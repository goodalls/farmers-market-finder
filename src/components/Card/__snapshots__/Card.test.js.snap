// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    markets={
        Array [
            Object {
              "Products": "hello; big; string",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          onClick={undefined}
>
          ♥
</span>,
        <h2
          className="name"
/>,
        <div
          className="info"
>
          <p>
                    Address: 
          </p>
          <p>
                    Schedule: 
          </p>
          <ol>
                    Products:
                    <li>
                              hello
                    </li>
                    <li>
                               big
                    </li>
                    <li>
                               string
                    </li>
          </ol>
</div>,
        <div
          id="map"
>
          <MapContainer />
</div>,
      ],
      "className": "text-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "♥",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "♥",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "name",
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Address: 
</p>,
            <p>
              Schedule: 
</p>,
            <ol>
              Products:
              <li>
                            hello
              </li>
              <li>
                             big
              </li>
              <li>
                             string
              </li>
</ol>,
          ],
          "className": "info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Address: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Address: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Schedule: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Schedule: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Products:",
                Array [
                  <li>
                    hello
</li>,
                  <li>
                     big
</li>,
                  <li>
                     string
</li>,
                ],
              ],
            },
            "ref": null,
            "rendered": Array [
              "Products:",
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": "hello",
                },
                "ref": null,
                "rendered": "hello",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": " big",
                },
                "ref": null,
                "rendered": " big",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": " string",
                },
                "ref": null,
                "rendered": " string",
                "type": "li",
              },
            ],
            "type": "ol",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MapContainer />,
          "id": "map",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            onClick={undefined}
>
            ♥
</span>,
          <h2
            className="name"
/>,
          <div
            className="info"
>
            <p>
                        Address: 
            </p>
            <p>
                        Schedule: 
            </p>
            <ol>
                        Products:
                        <li>
                                    hello
                        </li>
                        <li>
                                     big
                        </li>
                        <li>
                                     string
                        </li>
            </ol>
</div>,
          <div
            id="map"
>
            <MapContainer />
</div>,
        ],
        "className": "text-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "♥",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "♥",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "name",
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Address: 
</p>,
              <p>
                Schedule: 
</p>,
              <ol>
                Products:
                <li>
                                hello
                </li>
                <li>
                                 big
                </li>
                <li>
                                 string
                </li>
</ol>,
            ],
            "className": "info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Address: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Address: ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Schedule: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Schedule: ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Products:",
                  Array [
                    <li>
                      hello
</li>,
                    <li>
                       big
</li>,
                    <li>
                       string
</li>,
                  ],
                ],
              },
              "ref": null,
              "rendered": Array [
                "Products:",
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "hello",
                  },
                  "ref": null,
                  "rendered": "hello",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": " big",
                  },
                  "ref": null,
                  "rendered": " big",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": " string",
                  },
                  "ref": null,
                  "rendered": " string",
                  "type": "li",
                },
              ],
              "type": "ol",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MapContainer />,
            "id": "map",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
